agent/engine/dependencygraph/graph.go

```diff
@@ -377,6 +372,36 @@ func verifyContainerOrderingStatus(dependsOnContainer *apicontainer.Container) b
 		dependsOnContainerDesiredStatus == dependsOnContainer.GetSteadyStateStatus()
 }
 
+func verifyShutdownOrder(target *apicontainer.Container, existingContainers map[string]*apicontainer.Container) error {
+	// If the target is already terminal or if it doesn't need to be terminal, shutdown order doesn't apply
+	if !target.DesiredTerminal() || target.KnownTerminal() {
```

Comment:
should we move this check outside of this method? it's more logical to verify shutdown order only if desired status is terminal